client/src/LoginPage/LoginPage.jsx

```diff
@@ -0,0 +1,65 @@
+import React from 'react';
+import '../index.css';
+import { authenticationRepository, accountRepository } from '@/_services';
+
+class LoginPage extends React.Component {
+    constructor(props) {
+        super(props);
+
+        this.state = {
+            username: '',
+            password: ''
+        };
+
+        this.handleUsernameChange = this.handleUsernameChange.bind(this);
+        this.handlePasswordChange = this.handlePasswordChange.bind(this);
+        this.handleSubmit = this.handleSubmit.bind(this);
+    }
+
+    handleUsernameChange(event) {
+        this.setState({ username: event.target.value });
+    }
+
+    handlePasswordChange(event) {
+        this.setState({ password: event.target.value });
+    }
+
+    handleSubmit(event) {
+        event.preventDefault();
+
+        authenticationRepository.login(this.state.username.split('@')[0], this.state.password)
+            .then(() => {
```

Comment:
This is an improvement to your previous version, but I still don't get why we need to call `accountRepository.create()` here.

Would it be possible for `authenticationRepository.login` to return the ID of the account that is trying to log in?

Also it would be nice to have an error handler for `login` as entering wrong credentials is a common error.

Reply:
Right now, there's no exposure on the UI to create the account resource, so the client has to make it somewhere. 

Are you suggesting we make that resource when the user is logging in the `Login` API call? That right now returns the token. I should add Account ID in the response too?

Reply:
I think what confuses me is that we have create a resource at all. The account should already exist on the backend and we simply retrieve it here.

You seem to depend on three separate calls to do that:
1. Login and get an authentication token
2. "Create" a resource to get the ID of the account
3. Retrieve the account using the ID from step 2.

I think we should be able to remove at least one of those steps. I.e. the `login` API call could return the ID of our account. Another possibility would be to use the `get` API call without any ID and let the backend return the currently logged in account.

Or maybe I'm just understanding something incorrectly. Why exactly do we need to "create" / "make" that account resource first?

Reply:
I think I understand the issue: I'm assuming there can be multiple account resources, each one associated with a different admin, so that's why we have to make an account every time.

But it's true since we only really ever have one admin, our *GET* Account could be called without a guid query parameter, and get the default Account in the backend. But, that means we have to manually make that account through the API before we start the app. Would that be okay with you?

The Login API call should only ever return the token to be used for subsequent authorized API calls, and not the accountID though.

Reply:
Can we create that account when the backend starts up? So we don't need an extra API call for that.

If that takes too long, we can also keep it as-is for now, I was just wondering about the reason for this call

Reply:
That's a good idea, I can definitely make a default account on start up, and then just GET it in the client! Making changes.

Reply:
I've spent a few hours on this and it looks like it's too much scope unfortunately. I got an account on the server upon start up, but since there's no concept of account ID anymore (aka multiple accounts), all the other endpoints through all layers have to change. That, and integration tests and client have to change as well.

I'll keep it for now if that's okay!

Reply:
Of course. Don't worry about it too much, this application doesn't need to be perfect - we just want to see what tradeoffs were made and the thinking behind the architecture, etc. Your input and comments are very valuable.